store/src/java/com/zimbra/cs/account/AuthTokenUtil.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2018 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.cs.account;
+
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.AccountConstants;
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.common.soap.Element;
+import com.zimbra.common.util.StringUtil;
+import com.zimbra.common.util.ZimbraLog;
+
+public class AuthTokenUtil {
+
+    public static void encodeAuthResp(AuthToken at, Element parent, boolean isAdmin)  throws ServiceException {
+        if (isAdmin) {
```

Comment:
Find this slightly clearer:

        parent.addNonUniqueElement(
                isAdmin ? AdminConstants.E_AUTH_TOKEN :
                    AccountConstants.E_AUTH_TOKEN).setText(getOrigAuthData(at));

Reply:
done